pandas/core/groupby.py

```diff
@@ -2315,6 +2315,9 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                 # groupby
                 else:
                     cat = self.grouper.unique()
+                    cat.add_categories([c for c in self.grouper.categories
+                                        if c not in cat.categories],
```

Comment:
i like that better! (also vectorized).

as an aside, prob should add an asv for categorical grouping.